godot-macros/src/derive_godot_class.rs

```diff
@@ -176,17 +176,43 @@ struct ExportedField {
     field: Field,
     getter: String,
     setter: String,
+    hint: Option<ExportHint>,
+}
+
+#[derive(Clone)]
+struct ExportHint {
+    hint_type: Ident,
+    description: String,
+}
+
+impl ExportHint {
+    fn none() -> Self {
+        Self {
+            hint_type: ident("PROPERTY_HINT_NONE"),
+            description: "".to_string(),
+        }
+    }
 }
 
 impl ExportedField {
     pub fn new_from_kv(field: Field, parser: &mut KvParser) -> ParseResult<ExportedField> {
         let getter = parser.handle_lit_required("getter")?;
         let setter = parser.handle_lit_required("setter")?;
 
+        let hint = if let Some(hint_type) = parser.handle_ident("hint")? {
+            Some(ExportHint {
+                hint_type,
+                description: parser.handle_lit_required("hint_desc")?,
+            })
+        } else {
+            None
+        };
```

Comment:
Maybe you could use `map()` here?
```suggestion
        let hint = parser.handle_ident("hint")?
            .map(|hint_type| ExportHint {
                hint_type,
                description: parser.handle_lit_required("hint_desc")?,
            }).transpose()?
```

Not 100% sure if this is directly applicable, but [`transpose()`](https://doc.rust-lang.org/std/option/enum.Option.html#method.transpose) is quite a cool function 🙂